File: things/src/main/java/com/krisbiketeam/smarthomeraspbpi3/units/HomeUnitI2C.java
package com.krisbiketeam.smarthomeraspbpi3.units;

import android.support.annotation.NonNull;
import android.support.annotation.Nullable;

public class HomeUnitI2C extends HomeUnit {

    @Nullable
    @Override
    public Object readValue() {
        return null;
    }

    @Override
    public void registerListener(@NonNull HomeUnitListener listener) {

    }

    @Override
    public void unregisterListener() {

    }

    @Override
    public void close() {

    }
}

File: things/src/main/java/com/krisbiketeam/smarthomeraspbpi3/units/HomeUnitI2C.kt
package com.krisbiketeam.smarthomeraspbpi3.units

import timber.log.Timber
import java.io.IOException

interface HomeUnitI2C <T> : BaseUnit<T> {
    var device: AutoCloseable?

    override fun close() {
        try {
            device?.close()
        } catch (e: IOException) {
            Timber.e("Error closing PeripheralIO API on: $homeUnit", e)
        } finally {
            device = null
        }
    }

}
